### ISSUE TITLE
Feat: Add ARIA attributes to all forms

### ISSUE THREAD

**Issue by Avinash905** [MonalikaPatnaik/UMatter#1056]
### Describe the feature

ARIA attributes are crucial for enhancing web application accessibility, benefiting users with disabilities. Currently, the signin, signup, contact us, feedback forms lack these attributes, hindering proper interpretation by assistive technologies.

**Some steps to implement this enhancement-**

- Identify the form elements in your codebase.
- Determine the relevant ARIA attributes to add for each form element.
- Add ARIA roles to the forms and their elements (e.g., "form" role for the main form element).
- Include concise and descriptive ARIA labels using the aria-label attribute.
- Use the aria-describedby attribute to provide additional instructions for the forms.
- Implement ARIA states and properties (e.g., aria-required for required fields).

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'23 contributor
- [X] I want to work on this issue

**Comment by Avinash905** [comment]
Please assign me this issue so that I can work on this issue under **GSSOC'23**.
